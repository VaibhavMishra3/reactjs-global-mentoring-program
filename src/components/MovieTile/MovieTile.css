.movie-tile {
  position: relative;
  margin: 15px;
  width: 200px;
  background-color: #232323;
  border-radius: 8px;
  overflow: hidden;
  transition: transform 0.2s;
}

.movie-tile:hover {
  transform: scale(1.05);
}

.movie-poster {
  width: 100%;
  height: auto;
}

.movie-info {
  padding: 10px;
  color: white;
}

.three-dots {
  position: absolute;
  top: 10px;
  right: 10px;
  color: white;
  font-size: 24px;
  cursor: pointer;
}

.context-menu {
  position: absolute;
  top: 30px;
  right: 10px;
  background-color: #424242;
  border-radius: 4px;
  padding: 10px;
  color: white;
  z-index: 10;
}

.menu-item {
  padding: 5px 10px;
  cursor: pointer;
}

.menu-item:hover {
  background-color: #555555;
}
