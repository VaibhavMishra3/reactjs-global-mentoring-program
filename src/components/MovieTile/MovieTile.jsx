import React from 'react';
import { PropTypes } from 'prop-types';
import { useNavigate, useSearchParams } from 'react-router-dom';

import { formatGenres, formatYear } from '../../utils/formatters.js';
import { movieType } from '../../constants/types.js';
import { navigation } from '../../utils/navigation.js';
import posterNotAvailable from '../../assets/images/poster-not-available.jpg';

const MovieTile = ({ movie, handleClick }) => {
  const [searchParams] = useSearchParams();
  const navigate = useNavigate();

  const handleImageError = ({ target }) => {
    target.src = posterNotAvailable;
    target.alt = 'Poster not available';
  };

  return (
    <div data-testid='movie-tile' className='border'>
      <span
        data-testid='edit-movie'
        className='float-end'
        onClick={() => navigate(navigation(`/${movie.id}/edit`, searchParams))}
      >
        Edit
      </span>
      <div onClick={() => handleClick(movie)}>
        <div>
          <img
            data-testid='movie-tile-poster'
            className='img-fluid'
            src={movie.poster_path}
            alt={movie.title}
            onError={handleImageError}
          />
        </div>
        <div>
          <div>
            <span data-testid='movie-tile-title' className='h5 float-begin'>{movie.title}</span>
            {movie.release_date && <span data-testid='movie-tile-year' className='h5 m-2 border border-2 rounded float-begin'>{formatYear(movie.release_date)}</span>}
          </div>
          {movie.genres && <div data-testid='movie-tile-genres'>{formatGenres(movie.genres)}</div>}
        </div>
      </div>
    </div>
  )
}

MovieTile.propTypes = {
  movie: movieType.isRequired,
  handleClick: PropTypes.func.isRequired
};

export default MovieTile;
