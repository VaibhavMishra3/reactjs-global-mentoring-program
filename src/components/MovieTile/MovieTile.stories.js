import MovieTile from './MovieTile';
import { MOCK_MOVIES } from '../../constants/mockData.js';

export default {
  component: MovieTile
};

export const Default = {
  args: {
    movie: MOCK_MOVIES[0],
  },
  parameters: {
  }
};
