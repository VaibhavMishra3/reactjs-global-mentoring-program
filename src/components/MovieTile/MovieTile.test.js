import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { cleanup, render, screen } from '@testing-library/react'

import MovieTile from './MovieTile';
import { MOCK_MOVIES } from '../../constants/mockData.js';

describe('MovieTile', () => {

  afterEach(cleanup);

  it('should match snapshot', () => {
    const movieTile = renderer
      .create(
        <MemoryRouter>
          <MovieTile
            movie={MOCK_MOVIES[0]}
            handleClick={console.log}
          />
        </MemoryRouter>
      )
      .toJSON();

    expect(movieTile).toMatchSnapshot();
  });

  it('render correctly', () => {
    const movie = MOCK_MOVIES[0];

    render(
      <MemoryRouter>
        <MovieTile
          movie={movie}
          handleClick={console.log}
        />
      </MemoryRouter>
    );

    const image = screen.getByTestId('movie-tile-poster');
    expect(image.getAttribute('src')).toEqual(movie.poster_path);
    expect(image.getAttribute('alt')).toEqual(movie.title);

    expect(screen.getByTestId('movie-tile-title')).toHaveTextContent(movie.title);
    expect(screen.getByTestId('movie-tile-year')).toHaveTextContent('1994');
    expect(screen.getByTestId('movie-tile-genres')).toHaveTextContent('Thriller, Crime');
  });

});
