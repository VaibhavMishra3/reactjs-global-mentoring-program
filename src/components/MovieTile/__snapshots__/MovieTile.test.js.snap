// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile should match snapshot 1`] = `
<div
  className="border"
  data-testid="movie-tile"
>
  <span
    className="float-end"
    data-testid="edit-movie"
    onClick={[Function]}
  >
    Edit
  </span>
  <div
    onClick={[Function]}
  >
    <div>
      <img
        alt="Pulp Fiction"
        className="img-fluid"
        data-testid="movie-tile-poster"
        onError={[Function]}
        src="https://cdn.europosters.eu/image/1300/posters/pulp-fiction-cover-i1288.jpg"
      />
    </div>
    <div>
      <div>
        <span
          className="h5 float-begin"
          data-testid="movie-tile-title"
        >
          Pulp Fiction
        </span>
        <span
          className="h5 m-2 border border-2 rounded float-begin"
          data-testid="movie-tile-year"
        >
          1994
        </span>
      </div>
      <div
        data-testid="movie-tile-genres"
      >
        Thriller, Crime
      </div>
    </div>
  </div>
</div>
`;
